hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -568,29 +579,15 @@ public long getPurgeIntervalNanos() {
    * @param methodName - RPC Request method name
    * @param details - Processing Detail.
    *
-   * if this request took too much time relative to other requests
-   * we consider that as a slow RPC. 3 is a magic number that comes
-   * from 3 sigma deviation. A very simple explanation can be found
-   * by searching for 68-95-99.7 rule. We flag an RPC as slow RPC
-   * if and only if it falls above 99.7% of requests. We start this logic
-   * only once we have enough sample size.
+   * if this request took time exceed `logSlowRPCThresholdMs`
+   * we consider that as a slow RPC.
    */
   void logSlowRpcCalls(String methodName, Call call,
       ProcessingDetails details) {
-    final int deviation = 3;
-
-    // 1024 for minSampleSize just a guess -- not a number computed based on
-    // sample size analysis. It is chosen with the hope that this
-    // number is high enough to avoid spurious logging, yet useful
-    // in practice.
-    final int minSampleSize = 1024;
-    final double threeSigma = rpcMetrics.getProcessingMean() +
-        (rpcMetrics.getProcessingStdDev() * deviation);
 
     long processingTime =
             details.get(Timing.PROCESSING, rpcMetrics.getMetricsTimeUnit());
-    if ((rpcMetrics.getProcessingSampleCount() > minSampleSize) &&
-        (processingTime > threeSigma)) {
+    if (processingTime > getLogSlowRPCThresholdMs()) {
```

Comment:
We should keep both methods, rather than replacing the old one with the new one. We should do something like the following. 
```
long processingTimeThreshold = LONG.MAX;
if (slow_rpc_based_on_hard_threshold) {
   processingTimeThreshold = thresholdFromConfig 
} else {
   processingTimeThreshold = threeSigma;
}

if (processingTime > processingTimeThreshold) {
  ....
}
```

Reply:
I would also consider a different approach. 

```
    if ((rpcMetrics.getProcessingSampleCount() > minSampleSize) &&
        (processingTime > threeSigma) 
+       && (processingTime > minSlowRPCThreshold) ) {
        ....
    }
```

Basically, for a RPC request to be slow, its processing time has to meet the minSlowRPCThreshold and is an outlier.

Reply:
Thanks @xinglin  for your review, I learned from it.
In response to your suggestion, I prefer the second approach and consider to update the PR based on the second approach, thanks again.